source/MaterialXRuntime/RtNode.cpp

```diff
@@ -17,6 +17,28 @@ namespace MaterialX
 namespace
 {
     static const RtToken NODEDEF("nodedef");
+
+    static const RtToken XPOS_METADATA("xpos");
+    static const RtToken YPOS_METADATA("ypos");
+    static const RtToken WIDTH_METADATA("width");
+    static const RtToken HEIGHT_METADATA("height");
+    static const RtToken UI_COLOR_METADATA("uicolor");
+    static const RtToken UI_VISIBLE_METADATA("uivisible");
+    static const RtToken UI_ADVANCED_METADATA("uiadvanced");
+    static const RtToken FILE_PREFIX_METADATA("fileprefix");
+    static const RtToken COLOR_SPACE_METADATA("colorspace");
+    static const RtToken UNIT_METADATA("unit");
+    static const RtToken UNIT_TYPE_METADATA("unittype");
+    static const RtTokenVec INPUT_COLOR_PUBLIC_METADATA { COLOR_SPACE_METADATA };
+    static const RtTokenVec INPUT_FLOAT_PUBLIC_METADATA { UNIT_METADATA, UNIT_TYPE_METADATA };
+    static const RtTokenVec INPUT_PUBLIC_METADATA;
+    static const RtTokenVec PUBLIC_METADATA { XPOS_METADATA,
+                                              YPOS_METADATA,
+                                              WIDTH_METADATA,
+                                              HEIGHT_METADATA,
+                                              UI_COLOR_METADATA,
+                                              UI_VISIBLE_METADATA,
+                                              UI_ADVANCED_METADATA };
```

Comment:
I think there is a risk of static initialization order failure here. So to be 100% sure it's better to just define the tokens directly in the vector initialization lists. Like this:
```
static const RtTokenVec PUBLIC_METADATA { 
                                              RtToken("xpos"),
                                              RtToken("ypos"),
                                              RtToken("width"),
                                              ...
                                              };
```